README.md

```diff
@@ -1,2 +1,127 @@
+    s3://example-bucket/outputs/foobar_1.trimming_report.txt
+    s3://example-bucket/outputs/foobar_2.trimming_report.txt
+    s3://example-bucket/outputs/fastqc/
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.html
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.zip
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.html
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.zip
+    ```
+
+2. Prepare your Synapse configuration file to authenticate the workflow. For more details, check out the [Authentication](#authentication) section.
+
+    **Example:** Uploaded to `s3://example-bucket/synapse_config.ini`
+
+    ```ini
+    [authentication]
+    authtoken = <personal-access-token>
+    ```
+
+3. Create a parent Synapse folder that will contain the indexed files and the associated folder structure.
+
+    **Example:** Created a folder (`syn26601236`) under some project
+
+    ```text
+    synapse create --parentid <project-id> --name <folder-name> Folder
+    ```
+
+4. Prepare your parameters file. For more details, check out the [Parameters](#parameters) section.
+
+    **Example:** Stored locally as `./params.yml`
+
+    ```yaml
+    s3_prefix: s3://example-bucket/outputs/
+    synapse_config: s3://example-bucket/synapse_config.ini
+    parent_id: syn26601236
+    ```
+
+5. Launch workflow using the [Nextflow CLI](https://nextflow.io/docs/latest/cli.html#run), the [Tower CLI](https://help.tower.nf/latest/cli/), or the [Tower web UI](https://help.tower.nf/latest/launch/launchpad/).
+
+    **Example:** Launched using the Tower CLI
+
+    ```console
+    tw launch sage-bionetworks-workflows/nf-synindex --params-file=./params.yml
+    ```
+
+6. Explore the parent folder on Synapse.
+
+    **Example:** Synapse files and folders under `syn26601236`
+
+    ```text
+    syn26601270  foobar_1.trimming_report.txt
+    syn26601272  foobar_2.trimming_report.txt
+    syn26601252  fastqc/
+    syn26601268    foobar_1_val_1_fastqc.html
+    syn26601267    foobar_1_val_1_fastqc.zip
+    syn26601271    foobar_2_val_2_fastqc.html
+    syn26601269    foobar_2_val_2_fastqc.zip
+    ```
+
+7. Explore the output file mapping the S3 objects and the corresponding Synapse file IDs.
+
+    **Example:** Downloaded from `s3://example-bucket/outputs/synindex/under-syn26601236/file_ids.csv`
+
+    ```text
+    s3://example-bucket/outputs/foobar_1.trimming_report.txt,syn26601270
+    s3://example-bucket/outputs/foobar_2.trimming_report.txt,syn26601272
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.html,syn26601268
+    s3://example-bucket/outputs/fastqc/foobar_1_val_1_fastqc.zip,syn26601267
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.html,syn26601271
+    s3://example-bucket/outputs/fastqc/foobar_2_val_2_fastqc.zip,syn26601269
+    ```
+
+## Authentication
+
+Downloading files from Synapse requires the workflow to be authenticated. The workflow currently supports two authentication methods:
```

Comment:
I'm sure I'm missing something obvious here, but isn't the purpose of this workflow to take Nextflow output files that are not on Synapse, and index them in Synapse to make them available there? In what situations would this workflow be retrieving files from Synapse?

Reply:
Good catch! This is a copy-paste typo. I wrote the README for synstage first and didn't realize that I used a pipeline-specific term in the Authentication section.